src/pages/workspace/WorkspaceEditorPage.js

```diff
@@ -50,26 +51,35 @@ class WorkspaceEditorPage extends React.Component {
     }
 
     onImageSelected(image) {
+        updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: true});
         this.setState({previewAvatarURL: image.uri});
 
         // Store the upload avatar promise so we can wait for it to finish before updating the policy
         this.uploadAvatarPromise = uploadAvatar(image).then(url => new Promise((resolve) => {
+            updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: false});
             this.setState({avatarURL: url}, resolve);
-        }));
+        })).catch(() => {
+            Growl.error(this.props.translate('workspace.editor.avatarUploadFailureMessage'));
+            updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: false});
+        });
     }
 
     onImageRemoved() {
         this.setState({previewAvatarURL: '', avatarURL: ''});
     }
 
     submit() {
+        updateLocalPolicyValues(this.props.policy.id, {isPolicyUpdating: true});
+
         // Wait for the upload avatar promise to finish before updating the policy
         this.uploadAvatarPromise.then(() => {
             const name = this.state.name.trim();
             const avatarURL = this.state.avatarURL;
             const policyID = this.props.policy.id;
 
             update(policyID, {name, avatarURL});
+        }).catch(() => {
+            updateLocalPolicyValues(this.props.policy.id, {isPolicyUpdating: false});
```

Comment:
Since there isn't one of these in the `then()` how does the `isPolicyUpdating` get changed when the promise does not fail?

Reply:
It's in the `then` block of `update` function in the API call.

Reply:
Gotcha 👍